hardware_interface/test/test_resource_manager.cpp

```diff
@@ -74,13 +74,27 @@ TEST_F(TestResourceManager, initialization_with_urdf_manual_validation) {
 }
 
 TEST_F(TestResourceManager, initialization_with_wrong_urdf) {
-  try {
-    hardware_interface::ResourceManager rm(
-      ros2_control_test_assets::minimal_robot_missing_keys_urdf);
-    FAIL();
-  } catch (const std::exception & e) {
-    std::cout << e.what() << std::endl;
-    SUCCEED() << e.what();
+  // missing state keys
+  {
+    try {
```

Comment:
Wouldn't be simpler to use something like:

```
hardware_interface::ResourceManager rm();
EXPECT_THROW(rm.load_urdf(ros2_control_test_assets::minimal_robot_missing_state_keys_urdf), std::exception);
```

Reply:
you wouldn't be able to visualie the exception message in this case, right?

Reply:
Why do you want to visualize the message in a test? Let's just put an actual check on them, we know the message we should be seeing (or parts of it...)

Reply:
Yeah I agree with @bmagyar we don't need and message visualization because it should be clear from the test where the error is or could be.

Reply:
done in ed49b97